### PR TITLE
APE 17 migration

### PR DESCRIPTION
**Pull request by epassaro**
**Description**

Adoption of the new Astropy package template: https://docs.astropy.org/projects/package-template/en/latest/ape17.html

**What's working?**

- [ ] Package installation :warning: requires releasing manually a new compatible tag to work
- [x] Unit Tests
- [x] Documentation Build: https://epassaro.github.io/tardis/branch/ape-17-migration/installation.html#requirements
- [ ] Compare Reference Data (requires package installation)
- [x] Scripts/Entry points
- [x] CalVer support :)
 
Suggestions?

**Full list of changes**
1. ` .github/workflows/documentation-build.yml` - use "new" doc build command (deprecated `python setup.py build_docs`)
2.  `.github/workflows/documentation-preview.yml` - same as above
3. `.gitignore` - re-rendered with `cookiecutter` and replaced
4. `.gitmodules` - removed, `astropy_helpers` deprecated :)
5. `MANIFEST.in` - re-rendered
6. `README.md` - merged with file rendered with `cookiecutter`
7. `ah_bootstrap.py` - removed
8. `astropy_helpers` - removed
9. `docs/Makefile` - re-rendered
10. `docs/conf.py` - merged
11. `docs/make.bat` - re-rendered
12. `ez_setup.py` - removed
13. `licenses/LICENSE.rst` - moved to new `licenses` folder
14. `licenses/README.rst` - new
15. `licenses/TEMPLATE_LICENCE.rst` - new
16. `pyproject.toml` - merged
17.  `setup.cfg` - re-rendered, added `entry_points`, `package_data` and disabled `doctest`
18. `setup.py` - re-rendered, tweaked to make CalVer work
19. `tardis/__init__.py` - merged
20. `tardis/__astropy_init.py` - re-rendered
21. `tardis/conftest.py` - merged
22. `tardis/io/setup_package.py` - APE 17 step 4
23. `tardis/montecarlo/setup_package.py` - APE 17 step 4
24. `tardis/plasma/setup_package.py` -  APE 17 step 4
25. `tardis/simulation/setup_package.py` - APE 17 step 4
26. `scripts/tardis → tardis/scripts/tardis.py` - moved to make it work as an entry point
27. `tardis/tests/integration_tests/conftest.py` - `githash` not available in new version system (`setuptools_scm`)
28. `tardis/tests/integration_tests/plot_helpers.py ` - same as above
29. `tardis/tests/integration_tests/report.py` - same as above
30. `tardis/tests/setup_package.py` - APE17 step 4
31. `tardis_env3.yml` - add required `setuptools_scm` and `sphinx-astropy`
32. `tox.ini` - new from cookiecutter

**Coverage report**
Moving `tardis` script to `tardis/scripts` makes coverage fail (we have more code lines in PR than in the base branch)

**Motivation and context**

- Open issues:
  - #1211 
  - #1216
- Getting rid of troublesome `astropy_helpers` submodule
- Update `conda-forge` package

**How has this been tested?**
- [x] Testing pipeline.
- [x] Other. 

**Examples**

**Type of change**

- [ ] Bug fix. 
- [ ] New feature. 
- [x] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [ ] My change requires a change to the documentation.
    - [ ] I have updated the documentation accordingly.
    - [ ] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [ ] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

File: tardis/tests/integration_tests/plot_helpers.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 import tempfile
 
 from pytest_html import extras
-from tardis import __githash__ as tardis_githash
+
+# from tardis import __githash__ as tardis_githash
```
Comment (andrewfullard) [inline tardis/tests/integration_tests/plot_helpers.py:6]: Why comment and not just delete?

File: tardis/tests/integration_tests/report.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 # For specifying error while exception handling
 from socket import gaierror
 
-from tardis import __githash__ as tardis_githash
+# from tardis import __githash__ as tardis_githash
```
Comment (andrewfullard) [inline tardis/tests/integration_tests/report.py:36]: Why comment?

**Review by andrewfullard — COMMENTED** [review]
Some minor comments, otherwise looks good

File: tardis/tests/integration_tests/plot_helpers.py (hunk 1 above)
Comment (epassaro) [reply tardis/tests/integration_tests/plot_helpers.py:6]: Someone will need to fix integration tests in the future and maybe that line would be useful. I was thinking of adding a comment like `not available in version.py anymore, find another way!` or something like that.

File: licenses/LICENSE.rst [hunk 3]
```diff
@@ -0,0 +1,26 @@
+Copyright 2021 TARDIS Collaboration
```
[the remark this answers is not in the document]
Comment (epassaro) [reply licenses/LICENSE.rst:1]: Didn't see that. TARDIS is 2014-present or 2013-present? @wkerzendorf

[the remark this answers is not in the document]
Comment (epassaro) [reply licenses/LICENSE.rst:1]: Sorry, I thought we were talking about `conf.py`. We should put `Copyright 2014 TARDIS Collaboration` I guess.

File: tardis/scripts/tardis.py [hunk 4]
```diff
@@ -10,56 +10,70 @@
...
+)
+
+parser.add_argument(
+    "--packet_log_file",
+    default=None,
+    help="Name of the packet log file. Packet logging needs to be switched on before "
+    "compiling.",
+)
+
+parser.add_argument(
```
Comment (wkerzendorf) [inline tardis/scripts/tardis.py:41]: @KevinCawley what is that?

Comment (KevinCawley) [reply tardis/scripts/tardis.py:41]: I'm sorry?

**Pull request merged**